src/ripple/app/main/Application.cpp

```diff
@@ -824,24 +824,145 @@ class ApplicationImp
     beast::Journal journal (std::string const& name) override;
 
     //--------------------------------------------------------------------------
-    bool initSqliteDbs ()
+
+    bool
+    initDBs()
     {
+        // Initialize SQLite databases
         assert (mTxnDB.get () == nullptr);
         assert (mLedgerDB.get () == nullptr);
         assert (mWalletDB.get () == nullptr);
 
-        DatabaseCon::Setup setup = setup_DatabaseCon (*config_);
-        mTxnDB = std::make_unique <DatabaseCon> (setup, TxnDBName,
-                TxnDBInit, TxnDBCount);
-        mLedgerDB = std::make_unique <DatabaseCon> (setup, "ledger.db",
-                LedgerDBInit, LedgerDBCount);
-        mWalletDB = std::make_unique <DatabaseCon> (setup, "wallet.db",
-                WalletDBInit, WalletDBCount);
+        try
+        {
+            auto setup = setup_DatabaseCon(*config_);
+
+            // transaction database
+            mTxnDB = std::make_unique <DatabaseCon>(
+                setup,
+                TxnDBName,
+                TxnDBInit,
+                TxnDBCount);
+            mTxnDB->getSession() <<
+                boost::str(boost::format("PRAGMA cache_size=-%d;") %
+                (config_->getSize(siTxnDBCache) * kilobytes(1)));
+            mTxnDB->setupCheckpointing(m_jobQueue.get(), logs());
+
+            if (!setup.standAlone ||
+                setup.startUp == Config::LOAD ||
+                setup.startUp == Config::LOAD_FILE ||
+                setup.startUp == Config::REPLAY)
+            {
+                // perform any needed table updates
+                updateTxnDB();
+
+                // Check if AccountTransactions has primary key
```

Comment:
Why does it matter if AccountTransactions doesn't have a primary key? Why specifically check for this schema characteristic?

Reply:
I don't know the reason, its what rippled currently does.

Reply:
It would be probably a bit expensive to list all transactions by the Bitstamp issuing account without an index I guess.